tests/test_grpc_functions.py

```diff
@@ -0,0 +1,248 @@
+
+        self.assertEquals(0, len(expected))
+
+    def test_concat(self):
+        port = find_free_port()
+        env = {
+            'PYTHONPATH': '%s/tests/functions:$PYTHONPATH' % os.getcwd(),
+            'GRPC_PORT': str(port),
+            'FUNCTION_URI': 'file://%s/tests/functions/concat.py?handler=concat' % os.getcwd()
+        }
+
+        self.process = subprocess.Popen(self.command,
+                                        cwd=self.workingdir,
+                                        shell=True,
+                                        env=env,
+                                        preexec_fn=os.setsid,
+                                        )
+        time.sleep(0.5)
+
+        channel = grpc.insecure_channel('localhost:%s' % port)
+        self.stub = function.MessageFunctionStub(channel)
+
+        def generate_messages():
+            headers = {
+                'Content-Type': message.Message.HeaderValue(values=['application/json']),
+                'correlationId': message.Message.HeaderValue(values=[str(uuid.uuid4())])
+            }
+
+            messages = [
+                message.Message(payload='{"foo":"bar","hello":"world"}', headers=headers),
+            ]
+            for msg in messages:
+                yield msg
+
+        responses = self.stub.Call(generate_messages())
+
+        for response in responses:
+            self.assertEquals('{"result": "foobarhelloworld"}', response.payload)
+
+    def test_accepts_application_json(self):
+        port = find_free_port()
+        env = {
+            'PYTHONPATH': '%s/tests/functions:$PYTHONPATH' % os.getcwd(),
+            'GRPC_PORT': str(port),
+            'FUNCTION_URI': 'file://%s/tests/functions/concat.py?handler=concat' % os.getcwd()
+        }
+
+        self.process = subprocess.Popen(self.command,
+                                        cwd=self.workingdir,
+                                        shell=True,
+                                        env=env,
+                                        preexec_fn=os.setsid,
+                                        )
+        time.sleep(0.5)
+
+        channel = grpc.insecure_channel('localhost:%d' % port)
+        self.stub = function.MessageFunctionStub(channel)
+
+        def generate_messages():
+            headers = {
+                'Content-Type': message.Message.HeaderValue(values=['application/json']),
+                'Accept': message.Message.HeaderValue(values=['application/json']),
+                'correlationId': message.Message.HeaderValue(values=[str(uuid.uuid4())])
+            }
+
+            messages = [
+                message.Message(payload='{"foo":"bar","hello":"world"}', headers=headers),
+            ]
+            for msg in messages:
+                yield msg
+
+        responses = self.stub.Call(generate_messages())
+
+        for response in responses:
+            self.assertEquals('{"result": "foobarhelloworld"}', response.payload)
```

Comment:
Since the accept value is 'application/json' and the response isn't a dict, isn't the response marshalling supposed to fail?

Reply:
the response is a dict